diff --git a/www/sites/all/modules/ds/views/views_plugin_ds_fields_view.inc b/www/sites/all/modules/ds/views/views_plugin_ds_fields_view.inc
index c570e91..d457f81 100644
--- a/www/sites/all/modules/ds/views/views_plugin_ds_fields_view.inc
+++ b/www/sites/all/modules/ds/views/views_plugin_ds_fields_view.inc
@@ -41,6 +41,7 @@ class views_plugin_ds_fields_view extends views_plugin_row {
       'commerce_product' => 'product_id',
       'commerce_line_item' => 'line_item_id',
       'civicrm_event' => 'id',
+      'field_collection_item' => 'item_id',
     );
     return isset($base_table_fields[$this->base_table]) ? $base_table_fields[$this->base_table] : 'nid';
   }
